internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go

```diff
@@ -53,7 +55,7 @@ type Downloader struct {
 
 // NewDownloader creates and configures Elastic Downloader
 func NewDownloader(log *logger.Logger, config *artifact.Config, upgradeDetails *details.Details) (*Downloader, error) {
-	client, err := config.HTTPTransportSettings.Client(
+	client, err := config.Client(
```

Comment:
Was this change necessary for this PR?  If not, please revert it.  If it's just cleanup/refactoring, please do that in a separate PR.  We have this documented in our contributing guidelines as well: https://github.com/elastic/elastic-agent/blob/main/CONTRIBUTING.md#as-a-reviewee-ie-author-of-a-pr.

Same for the similar change below as well.